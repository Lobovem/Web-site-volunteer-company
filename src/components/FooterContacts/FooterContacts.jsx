import s from './FooterContacts.module.scss';

import phone from '../../img/icon-phone.svg';
import adress from '../../img/icon-adress.svg';

export const FooterContacts = () => {
  return (
    <div className={s.contacts}>
      <div className={s.contacts__wrap}>
        <h2 className={s.contacts__title}>контакти</h2>
        <div className={s.contacts__phoneBox}>
          <img className={s.contacts__img} src={phone} alt="icon-phone.svg" />
          <a className={s.contacts__desc} href="tel:+380630494002">
            +380 (63)-049-40-02
          </a>
          <a className={s.contacts__desc} href="tel:+380671597373">
            +380 (67)-159-73-73
          </a>
        </div>

        <div className={s.contacts__adressBox}>
          <img className={s.contacts__img} src={adress} alt="icon-adress.svg" />
          <adress className={s.contacts__desc}>м. Харків, вул.Космічна, буд. 21, оф. 529</adress>
        </div>
      </div>
    </div>
  );
};
