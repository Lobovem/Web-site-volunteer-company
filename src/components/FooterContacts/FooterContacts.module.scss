@import './src/styles/MixinsMedia.module.scss';
@import './src/styles/Variables.module.scss';
@import './src/styles/MixinsLayout.module.scss';
@import './src/styles/MixinsKit.module.scss';

.contacts {

  flex: 1;

  &__wrap {
    margin-bottom: 50px;
  }

  &__title {
    font-weight: $font-weight-500;
    font-size: $font-size-24;
    line-height: 24px;
    text-transform: uppercase;
    color: $color-white;
    mix-blend-mode: normal;
    margin: 0;
    padding: 0;
    margin-bottom: 30px;
  }

  &__desc {
    font-weight: $font-weight-500;
    font-size: $font-size-14;
    line-height: 17px;
    color: $color-white;
    mix-blend-mode: normal;
    text-decoration: none;
    display: block;
    margin-bottom: 5px;
  }

  &__phoneBox {
    position: relative;
    margin-bottom: 20px;
    padding-left: 40px;
  }

  &__adressBox {
    position: relative;
    padding-left: 40px;

  }

  &__img {
    position: absolute;
    top: 50%;
    left: 0;
    transform: translate(0%, -50%);
  }
}