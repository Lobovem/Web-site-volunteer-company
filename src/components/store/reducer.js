import { createSlice } from '@reduxjs/toolkit';

const initialState = {
  burgerMenuState: false,
};

export const appReducer = createSlice({
  name: 'app',
  initialState: initialState,
  reducers: {
    isOpenBurgerMenu: (state, action) => {
      state.burgerMenuState = !state.burgerMenuState;
    },
  },
});

export const burgerMenuSelector = (state) => state.app.burgerMenuState;

export const { isOpenBurgerMenu } = appReducer.actions;
export default appReducer.reducer;
