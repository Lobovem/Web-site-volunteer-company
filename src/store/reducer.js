import { createSlice } from '@reduxjs/toolkit';

const initialState = {
  todoListState: [],
};

export const appReducer = createSlice({
  name: 'todoList',
  initialState: initialState,
  reducers: {
    // add_todo: (state, action) => {
    //   state.todoListState = [
    //     ...state.todoListState,
    //     {
    //       id: Math.random(),
    //       task: action.payload,
    //       complete: false,
    //       isEditing: false,
    //     },
    //   ];
    // },
    // delete_todo: (state, action) => {
    //   state.todoListState = state.todoListState.filter((todo) => todo.id !== action.payload.id);
    // },
  },
});

// export const userInputSelector = (state) => state.todoList.inputState;
// export const todoListSelector = (state) => state.todoList.todoListState;

// export const { add_todo } = todoReducer.actions;
export default appReducer.reducer;
