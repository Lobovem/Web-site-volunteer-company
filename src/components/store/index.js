// import configureStore from '@reduxjs/toolkit';
import { configureStore } from '@reduxjs/toolkit';
import { appReducer } from './reducer';

export const store = configureStore({
  reducer: {
    todoList: appReducer.reducer,
  },
});
