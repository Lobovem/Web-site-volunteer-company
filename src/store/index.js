// import configureStore from '@reduxjs/toolkit';
import { configureStore } from '@reduxjs/toolkit';
import { todoReducer } from './reducer';

export const store = configureStore({
  reducer: {
    todoList: todoReducer.reducer,
  },
});
