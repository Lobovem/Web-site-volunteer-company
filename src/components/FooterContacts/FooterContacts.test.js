import { render, screen } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import { FooterContacts } from './FooterContacts';
import { Provider } from 'react-redux';
import { store } from '../../redux/config/store';

describe('FooterContacts` component', () => {
  test('renders FooterContacts component', () => {
    // eslint-disable-next-line testing-library/render-result-naming-convention
    const component = render(
      <Provider store={store}>
        <Router>
          <FooterContacts />
        </Router>
      </Provider>
    );
    expect(component).toMatchSnapshot();
  });

  test('should render the title component contacts', () => {
    render(
      <Provider store={store}>
        <Router>
          <FooterContacts />
        </Router>
      </Provider>
    );
    const titleSection = screen.getByRole('heading', { name: /КОНТАКТИ/i });
    expect(titleSection).toBeInTheDocument();
  });

  test('renders and search elements links', () => {
    render(
      <Provider store={store}>
        <Router>
          <FooterContacts />
        </Router>
      </Provider>
    );
    const phoneLinks = screen.getAllByRole('link'); // Search all links
    expect(phoneLinks).toHaveLength(3);

    const phoneLink1 = screen.getByText('+380 (63)-049-40-02'); // Search link about text
    expect(phoneLink1).toBeInTheDocument(); // Check out link in DOM

    const phoneLink2 = screen.getByText('+380 (67)-159-73-73');
    expect(phoneLink2).toBeInTheDocument();

    const addressLink = screen.getByRole('link', { name: /м\. Харків/i }); // Search link about text
    expect(addressLink).toBeInTheDocument(); // Check out link in DOM
  });

  test('renders FooterContacts component with 2 images (link elements)', () => {
    render(
      <Provider store={store}>
        <Router>
          <FooterContacts />
        </Router>
      </Provider>
    );
    const images = screen.getAllByRole('img');
    expect(images).toHaveLength(2);
  });
});
