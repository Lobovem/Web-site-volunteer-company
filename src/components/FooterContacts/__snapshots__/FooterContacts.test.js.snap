// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FooterContacts\` component renders FooterContacts component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="contacts"
      >
        <div
          class="contacts__wrap"
        >
          <h2
            class="contacts__title"
          >
            контакти
          </h2>
          <div
            class="contacts__phoneBox"
          >
            <img
              alt="icon-phone.svg"
              class="contacts__img"
              src="./img/icon-phone.svg"
            />
            <a
              class="contacts__link"
              href="tel:+380630494002"
            >
              +380 (63)-049-40-02
            </a>
            <a
              class="contacts__link"
              href="tel:+380671597373"
            >
              +380 (67)-159-73-73
            </a>
          </div>
          <div
            class="contacts__adressBox"
          >
            <img
              alt="icon-adress.svg"
              class="contacts__img"
              src="./img/icon-adress.svg"
            />
            <a
              class="contacts__link"
              href="https://goo.gl/maps/uXTuc6J6nSarrwsW7"
              target="blank"
            >
              <address
                class="contacts__desc"
              >
                м. Харків, вул.Космічна, буд. 21, оф. 529
              </address>
            </a>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="contacts"
    >
      <div
        class="contacts__wrap"
      >
        <h2
          class="contacts__title"
        >
          контакти
        </h2>
        <div
          class="contacts__phoneBox"
        >
          <img
            alt="icon-phone.svg"
            class="contacts__img"
            src="./img/icon-phone.svg"
          />
          <a
            class="contacts__link"
            href="tel:+380630494002"
          >
            +380 (63)-049-40-02
          </a>
          <a
            class="contacts__link"
            href="tel:+380671597373"
          >
            +380 (67)-159-73-73
          </a>
        </div>
        <div
          class="contacts__adressBox"
        >
          <img
            alt="icon-adress.svg"
            class="contacts__img"
            src="./img/icon-adress.svg"
          />
          <a
            class="contacts__link"
            href="https://goo.gl/maps/uXTuc6J6nSarrwsW7"
            target="blank"
          >
            <address
              class="contacts__desc"
            >
              м. Харків, вул.Космічна, буд. 21, оф. 529
            </address>
          </a>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
